### ISSUE TITLE
Why is fraternal-dropout more applicable to sequential tasks？

### ISSUE THREAD

**Issue by Maggione** [kondiz/fraternal-dropout#1]
Hello! I am really interested in your paper. And I have some questions about it.
Why is the algorithm designed specifically to address problems in RNNs? And it shows that the algorithm performs at par with II-model on the CIFAR-10 dataset, but leads to performance improvement by a significant margin in sequential tasks. I don't quite understand the reason about that.
Thank you very much!

**Comment by kondiz** [comment]
Thank you very much for your comment!

The fraternal dropout method is indeed general and may be applied in feed-forward architectures (as shown in the paper for CIFAR-10 semisupervised example). However, I believe that it is more powerful in the case of RNNs because:
1. Variance in prediction accumulates among time steps in RNNs and since you share parameters for all time steps, you may use the same kappa value at each step. In feed-forward networks the layers usually don't share parameters and hence you may want to use different kappa values for different layers (which may be hard to tune). The simple way to alleviate this problem is to apply the regularization term on the pre-softmax predictions only (as shown in the paper) or use the same kappa value for all layers. However, I believe that it may limit possible gains.
2. The best performing RNN architectures (state-of-the-art) usually use some kind of dropout (embedding dropout, word dropout, weight dropout etc.), very often with high dropout rates (even larger than 50% for input word embedding in NLP tasks). However, this is not true for feed-forward networks. For instance, ResNet architectures very often do not use dropout at all (probably because batch normalization is often better to use). You can see in the fraternal dropout paper (semisupervised CIFAR-10 task) that when unlabeled data is not used the regular dropout hurts performance and using fraternal dropout seems to improve just a little (it has to catch up what you lost by using dropout).

To conclude, I believe that when the use of dropout benefits in your architecture, applying fraternal dropout should improve performance even more.
Also in semisupervised setting, you may leverage unlabeled data by applying fraternal dropout (you do not have to know targets to apply regularization term).

As a final note, in image recognition tasks, you may experiment with something what I would temporarily dub 'fraternal augmentation'. Even though you do not have dropout, you use random data augmentation (random crop, random flip). Hence, you may force your neural network to have the same predictions for different augmentations.
